tests/Common/ConfigurationMetadataTest.php

```diff
@@ -0,0 +1,399 @@
+            ->setMetadata([
+                [
+                    'key' => 'someMetadataKey',
+                    'value' => 'development branch value',
+                ]
+            ]);
+        $branchComponents->addConfigurationMetadata($configurationMetadataOptions);
+
+        $branchComponents->resetToDefault('transformation', $configurationNameMain1);
+
+        $listConfigurationMetadata = $branchComponents->listConfigurationMetadata((new ListConfigurationMetadataOptions())
+            ->setComponentId('transformation')
+            ->setConfigurationId($configurationNameMain1));
+        self::assertCount(3, $listConfigurationMetadata);
+
+        $this->assertMetadataEquals(self::TEST_METADATA[0], $listConfigurationMetadata[0]);
+        $this->assertMetadataEquals($updatedMetadata[0], $listConfigurationMetadata[1]);
+        $this->assertMetadataEquals($moreMetadata[0], $listConfigurationMetadata[2]);
+    }
+
+    public function testConfigMetadataRestrictionsForReadOnlyUser()
+    {
+        $defaultBranchId = $this->getDefaultBranchId($this);
+
+        $guestClient = $this->getBranchAwareClient($defaultBranchId, [
+            'token' => STORAGE_API_READ_ONLY_TOKEN,
+            'url' => STORAGE_API_URL,
+            'backoffMaxTries' => 1,
+            'jobPollRetryDelay' => function () {
+                return 1;
+            },
+        ]);
+
+        $configurationNameMain1 = $this->generateUniqueNameForString('main-1');
+
+        // create new configurations in main branch
+        $components = new Components($this->_client);
+        $transformationMain1Options = (new Configuration())
+            ->setComponentId('transformation')
+            ->setConfigurationId($configurationNameMain1)
+            ->setName('Main 1')
+            ->setConfiguration(['test' => 'false']);
+
+        $components->addConfiguration($transformationMain1Options);
+        $configurationMetadataOptions = (new ConfigurationMetadata($transformationMain1Options))
+            ->setMetadata(self::TEST_METADATA);
+
+        try {
+            $components = new Components($guestClient);
+            $components->addConfigurationMetadata($configurationMetadataOptions);
+            $this->fail('should fail, insufficiently permission');
+        } catch (ClientException $e) {
+            $this->assertContains('Configuration manipulation is restricted for your user role', $e->getMessage());
+            $this->assertSame(403, $e->getCode());
+        }
+    }
+
+    private function assertMetadataEquals(array $expected, array $actual)
+    {
+        foreach ($expected as $key => $value) {
+            self::assertArrayHasKey($key, $actual);
+            self::assertSame($value, $actual[$key]);
+        }
+        self::assertArrayHasKey('timestamp', $actual);
+    }
+
+    private function createConfiguration($components, $componentId, $configurationId, $name = 'New Config')
```

Comment:
Je nějaký důvod proč by to mělo mít default value pro name? Budeš někdy chtít aby se ti vytvořil config s takhle generckým názvem?